controllers/session/session_controller_int_test.go

```diff
@@ -284,6 +284,103 @@ var _ = Describe("Complete session manipulation", func() {
 			})
 		})
 
+		Context("when calculating components readiness", func() {
+			It("should all show ready on success", func() {
+				req1 := reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      "test-session1",
+						Namespace: "test",
+					},
+				}
+				// Given - create first ref
+				res1, err := controller.Reconcile(context.Background(), req1)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(res1.Requeue).To(BeFalse())
+
+				// Then - all components should be success
+				session := get.Session("test", "test-session1")
+
+				Expect(*session.Status.State).To(Equal(v1alpha1.StateSuccess))
+				Expect(session.Status.Readiness.Components.Pending).To((HaveLen(0)))
+				Expect(session.Status.Readiness.Components.UnReady).To((HaveLen(0)))
+				Expect(session.Status.Readiness.Components.Ready).ToNot((HaveLen(0)))
+			})
+
+			It("should show pending before execution and missing modification, and unready on unsuccess", func() {
+				testLocator := func(ctx model.SessionContext, ref model.Ref, store model.LocatorStatusStore, report model.LocatorStatusReporter) error {
+					// When - three Resources are found
+					report(model.LocatorStatus{
+						Resource: model.Resource{
+							Namespace: ctx.Namespace,
+							Kind:      "Pod",
+							Name:      "TestPod",
+						},
+						Action: model.ActionLocated,
+					})
+					report(model.LocatorStatus{
+						Resource: model.Resource{
+							Namespace: ctx.Namespace,
+							Kind:      "Pod",
+							Name:      "TestPod2",
+						},
+						Action: model.ActionLocated,
+					})
+					report(model.LocatorStatus{
+						Resource: model.Resource{
+							Namespace: ctx.Namespace,
+							Kind:      "Pod",
+							Name:      "TestPod3",
+						},
+						Action: model.ActionLocated,
+					})
+
+					return nil
+				}
+				testRegistrar := func() (client.Object, model.Modificator) {
+					return &corev1.Pod{}, func(context model.SessionContext, ref model.Ref, store model.LocatorStatusStore, reporter model.ModificatorStatusReporter) {
+
+						// Then - Verify all are in pending stage before modification stage
+						session := get.Session("test", "test-session1")
+						Expect(session.Status.Readiness.Components.Pending).To((HaveLen(3)))
+
+						// When - One is reported success
+						reporter(model.ModificatorStatus{
+							LocatorStatus: store("Pod")[0],
+							Success:       false,
```

Comment:
`// When - One is reported success` but `Success: false`? Not sure I understand this.

Reply:
Same reversed logic vs comment in the next step.

Reply:
I guess it's passing because of weak assertions checking size of the slice vs actual content. Or am I missing sth here?

Reply:
Just comments that are in the wrong place. But yeah, it's checking sizes of things.

Reply:
fixed https://github.com/maistra/istio-workspace/pull/875/commits/269a7b34b1ac42c4a21d809f820880405e27cce5